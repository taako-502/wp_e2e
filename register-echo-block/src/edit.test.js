import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Edit from './edit'
import React from 'react'

// テストのセットアップ
global.React = React
Enzyme.configure({ adapter: new Adapter() })

test('テキストボックスがレンダリングされる', () => {
  const wrapper = shallow(<Edit attributes={{ text: 'こんにちは' }} />)
  const input = wrapper.find('input')
  expect(input).toHaveLength(1)
})

test('`attributes.text` で与えられた値がテキストボックスにセットされる', () => {
  const wrapper = shallow(<Edit attributes={{ text: 'こんにちは' }} />)
  const input = wrapper.find('input')
  expect(input.get(0).props.value).toBe('こんにちは')
})

test('input が変更されると`setAttributes`がコールされる', () => {
  const mockFn = jest.fn()
  const mockEvent = { target: { value: 'こんばんは' } }
  const wrapper = shallow(
    <Edit attributes={{ text: 'こんにちは' }} setAttributes={mockFn} />
  )
  const input = wrapper.find('input')
  input.simulate('change', mockEvent)
  expect(mockFn).toBeCalledWith({ text: 'こんばんは' })
})
