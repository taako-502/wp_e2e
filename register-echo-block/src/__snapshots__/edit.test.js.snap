// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショットが一致する 1`] = `
<div>
  <label
    htmlFor="echo-text"
  >
    Echo: 
  </label>
  <input
    id="echo-text"
    onChange={[Function]}
    type="text"
    value="こんにちは"
  />
</div>
`;
